xdsl/ir/core.py

```diff
@@ -456,6 +462,96 @@ def print_parameter(self, printer: Printer) -> None:
         """Print the attribute parameter."""
 
 
+class StrEnum(str, Enum):
+    """
+    Homemade StrEnum. StrEnum is standard in Python>=3.11.
```

Comment:
Can we add a deprecation warning if Python 3.11 and higher is used? It will help with version transition when we actually do so.

Reply:
I don't think that'll work, my current understanding is that we support a range of versions, including 3.10 and 3.11, and would like the deprecation warning to show in neither of them. I don't know if there's a way to warn only when we drop support for 3.10. I expect we'll do an audit of the codebase when that happens (in a year or two).

Reply:
OK then!